### PR TITLE
feat: support deletion of SCC Notification not in use in organization

### PR DESCRIPTION
**Pull request by daniel-cit**
This PR supports the deletion of Security Command Center Notification not in use in organization that match a given pattern

### CODE REVIEW

File: modules/project_cleanup/function_source/main.go [hunk 1]
```diff
@@ -338,6 +403,48 @@ func invoke(ctx context.Context) {
...
+		if p.LifecycleState == "DELETE_REQUESTED" {
+			return true
+		}
+		return false
+	}
+
+	removeSCCNotifications := func(organization string) {
+		logger.Printf("Try to remove SCC Notifications from organization [%s]", organization)
+		req := &securitycenterpb.ListNotificationConfigsRequest{
+			Parent: fmt.Sprintf("organizations/%s", organization),
```
Comment (bharathkkb) [inline modules/project_cleanup/function_source/main.go:421]: It looks like default it returns 10 but configurable to 1000? Could we request a higher number?

Comment (bharathkkb) [reply modules/project_cleanup/function_source/main.go:421]: https://pkg.go.dev/cloud.google.com/go/securitycenter/apiv1/securitycenterpb#ListNotificationConfigsRequest

**Review by bharathkkb — APPROVED** [review]
Thanks @daniel-cit a few minor comments

File: modules/project_cleanup/function_source/main.go (hunk 1 above)
Comment (daniel-cit) [reply modules/project_cleanup/function_source/main.go:421]: the quota is [500 per organization](https://cloud.google.com/security-command-center/quotas#notifications)
I will use this value for the page size

Comment (daniel-cit) [reply modules/project_cleanup/function_source/main.go]: fixed

File: modules/project_cleanup/function_source/main.go [hunk 2]
```diff
@@ -194,6 +213,29 @@ func getLabelsMapFromEnv(envVariableName string) map[string]string {
...
+	err := json.Unmarshal([]byte(targetExcludedSCCNotis), &sccNotis)
+	if err != nil {
+		logger.Printf("Failed to get SCC Notifications list from [%s] env variable, error [%s]", envVariableName, err.Error())
+	} else {
+		logger.Printf("Got SCC Notifications list [%s] from [%s] env variable", sccNotis, envVariableName)
+	}
+	//build Regexes
+	reg := make([]*regexp.Regexp, len(sccNotis))
+	for i, r := range sccNotis {
+		reg[i] = regexp.MustCompile(r)
```
[the remark this answers is not in the document]
Comment (daniel-cit) [reply modules/project_cleanup/function_source/main.go]: fixed

File: modules/project_cleanup/function_source/main.go [hunk 3]
```diff
@@ -194,6 +213,29 @@ func getLabelsMapFromEnv(envVariableName string) map[string]string {
...
+	logger.Println("Try to get SCC Notifications list")
+	if targetExcludedSCCNotis == "" {
+		logger.Printf("No SCC Notifications provided.")
+		return nil
+	}
+
+	var sccNotis []string
+	err := json.Unmarshal([]byte(targetExcludedSCCNotis), &sccNotis)
+	if err != nil {
+		logger.Printf("Failed to get SCC Notifications list from [%s] env variable, error [%s]", envVariableName, err.Error())
```
[the remark this answers is not in the document]
Comment (daniel-cit) [reply modules/project_cleanup/function_source/main.go:230]: fixed

File: modules/project_cleanup/function_source/main.go [hunk 4]
```diff
@@ -194,6 +213,29 @@ func getLabelsMapFromEnv(envVariableName string) map[string]string {
 	return labels
 }
 
+func getSCCNotiListFromEnv(envVariableName string) []*regexp.Regexp {
+	targetExcludedSCCNotis := os.Getenv(envVariableName)
```
[the remark this answers is not in the document]
Comment (daniel-cit) [reply modules/project_cleanup/function_source/main.go]: fixed

**Review by bharathkkb — APPROVED** [review]
@daniel-cit could you rebase?

**Comment by daniel-cit** [comment]
> @daniel-cit could you rebase?

Done

**Comment by daniel-cit** [comment]
@bharathkkb build is green 🟢 :)

**Pull request merged**